chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
+        "SMY": {'type': 'string', 'max_len': 8, 'charset': _SSTV_CHARSET},
+        "SQ 0": _SQUELCH,
+        "SQ 1": _SQUELCH,
+        "STAT 1": _STATUS_TEXT,
+        "STAT 2": _STATUS_TEXT,
+        "STAT 3": _STATUS_TEXT,
+        "STXR": {'type': 'list',
+                 'values': ("Off", "1/1", "1/2", "1/3", "1/4", "1/5", "1/6",
+                            "1/7", "1/8")},
+        "SV": {'type': 'list',
+               'values': ("Off", "0.2s", "0.4s", "0.6s", "0.8s", "1.0s",
+                          "2s", "3s", "4s", "5s")},
+        "TEMP": {'type': 'list',
+                 'values': ("°F", "°C")},
+        "TH": _BOOL,
+        "TNC": _BOOL,
+        "TSP": _BOOL,
+        "TXD": {'type': 'map',
+                'map': (("100ms", "1"), ("200ms", "2"), ("300ms", "3"),
+                        ("400ms", "4"), ("500ms", "5"), ("750ms", "6"),
+                        ("1000ms", "7"))},
+        "TXH": _BOOL,
+        "TXI": {'type': 'list',
+                'values': ("30sec", "1min", "2min", "3min", "4min", "5min",
+                           "10min", "20min", "30min")},
+        "TXS": _BOOL,
+        "TZ": {'type': 'list',
+               'values': ("UTC - 12:00", "UTC - 11:30", "UTC - 11:00",
+                          "UTC - 10:30", "UTC - 10:00", "UTC - 9:30",
+                          "UTC - 9:00", "UTC - 8:30", "UTC - 8:00",
+                          "UTC - 7:30", "UTC - 7:00", "UTC - 6:30",
+                          "UTC - 6:00", "UTC - 5:30", "UTC - 5:00",
+                          "UTC - 4:30", "UTC - 4:00", "UTC - 3:30",
+                          "UTC - 3:00", "UTC - 2:30", "UTC - 2:00",
+                          "UTC - 1:30", "UTC - 1:00", "UTC - 0:30",
+                          "UTC", "UTC + 0:30", "UTC + 1:00", "UTC + 1:30",
+                          "UTC + 2:00", "UTC + 2:30", "UTC + 3:00",
+                          "UTC + 3:30", "UTC + 4:00", "UTC + 4:30",
+                          "UTC + 5:00", "UTC + 5:30", "UTC + 6:00",
+                          "UTC + 6:30", "UTC + 7:00", "UTC + 7:30",
+                          "UTC + 8:00", "UTC + 8:30", "UTC + 9:00",
+                          "UTC + 9:30", "UTC + 10:00", "UTC + 10:30",
+                          "UTC + 11:00", "UTC + 11:30", "UTC + 12:00")},
+        "UPR": {'type': 'string', 'max_len': 9,
+                'charset': "ABCDEFGHIJKLMNOPQRSTUVWXYZ-0123456789"},
+        "VCS": _BOOL,
+        "WAY": {'type': 'list',
+                'values': ("Off", "6 digit NMEA", "7 digit NMEA",
+                           "8 digit NMEA", "9 digit NMEA", "6 digit Magellan",
+                           "DGPS")}}
+
+    _PROGRAMMABLE_VFOS = (
+        ("Band A, 118MHz Sub-Band", 1, 118, 135),
+        ("Band A, VHF Sub-Band", 2, 136, 173),
+        ("Band B, VHF Sub-Band", 3, 144, 147),
+        ("Band B, UHF Sub-Band", 6, 400, 479))
+
+    _baud = 9600
+    _cur_setting = 0
+    _memcache = {}
+    _setcache = {}
```

Comment:
It's a bad idea to do this (existing examples in the code notwithstanding). This is a mutable-type variable on the class, which is shared amongst all instances of this class until you do something like L552. better to leave these out and depend on the init code to create them so they never get accidentally shared.

Reply:
Sounds good, not being a python developer, these are the type of nuances I'm unaware of.

Reply:
Yep, it's not something I did a very good job of guarding against in the early days, despite being a python guy. It's obscure for sure, but learn from my scars :)

Reply:
Done.